numpy/core/tests/test_regression.py

```diff
@@ -2500,3 +2500,12 @@ def __len__(self):
         t = T()
         #gh-13659, would raise in broadcasting [x=t for x in result]
         np.array([t])
+
+    @pytest.mark.skipif(sys.maxsize < 2 ** 31 + 1, reason='overflows 32-bit python')
+    @pytest.mark.skipif(sys.platform == 'win32', reason='overflows on windows')
+    def test_to_ctypes(self):
+        #gh-14214
+        arr = np.zeros((2 ** 31 + 1,), 'b')
+        assert(arr.size * arr.itemsize > 2 ** 31)
```

Comment:
nit: parens here are weird.

Could use our `assert_(...)`, but probably best to drop them as `assert ...` and let pytest handle the message
```suggestion
        assert arr.size * arr.itemsize > 2 ** 31
```